GitUI/GitUIExtensions.cs

```diff
@@ -34,82 +34,90 @@ public static class GitUIExtensions
             return module.GetSingleDiff(firstRevision?.ToString(), secondRevision?.ToString(), file.Name, file.OldName, diffArgs, encoding, true, isTracked);
         }
 
-        [CanBeNull]
-        private static string GetSelectedPatch(
-            [NotNull] this FileViewer diffViewer,
-            [CanBeNull] ObjectId firstRevision,
-            [CanBeNull] ObjectId secondRevision,
-            [NotNull] GitItemStatus file)
+        /// <summary>
+        /// View the changes between the revisions, if possible as a diff
+        /// </summary>
+        /// <param name="fileViewer">Current FileViewer</param>
+        /// <param name="firstId">The first (A) commit</param>
+        /// <param name="selectedRev">The selected (B) commit</param>
+        /// <param name="file">The git item to view</param>
+        /// <param name="defaultText">default text if no diff is possible</param>
+        /// <param name="openWithDiffTool">The difftool command to open with</param>
+        /// <returns>Task to view</returns>
+        public static Task ViewChangesAsync(this FileViewer fileViewer,
+            [CanBeNull] ObjectId firstId,
+            [CanBeNull] GitRevision selectedRev,
+            [CanBeNull] GitItemStatus file,
+            [NotNull] string defaultText = "",
+            [CanBeNull] Action openWithDiffTool = null)
         {
-            if (!file.IsTracked)
+            if (file == null || selectedRev?.ObjectId == null)
             {
-                var fullPath = Path.Combine(diffViewer.Module.WorkingDir, file.Name);
-                if (Directory.Exists(fullPath) && GitModule.IsValidGitWorkingDir(fullPath))
+                if (defaultText.IsNotNullOrWhitespace())
                 {
-                    // git-status does not detect details for untracked and git-diff --no-index will not give info
-                    return LocalizationHelpers.GetSubmoduleText(diffViewer.Module, file.Name.TrimEnd('/'), "");
+                    return fileViewer.ViewTextAsync(file?.Name, defaultText, openWithDiffTool);
                 }
-            }
```

Comment:
Can the whole block `if (!file.IsTracked)` be omitted safely?

Reply:
Yes, !IsTracked -> IsNew is always set and we never try to get patches
This is handled in FileViewr.cs ViewFileAsync():397 (also for other code paths), see also #7869